### DISCUSSION TITLE
Does Mimir periodically delete data from the Object Storage (Minio)

### DISCUSSION THREAD

**Discussion by udeshyaG** [grafana/mimir#1938] [Help and support] [answered]
I have configured Mimir using the Getting started guide. I have configured Minio in distributed mode using 4 VMs and 8 drives as explained in [the docs](https://docs.min.io/minio/baremetal/installation/deploy-minio-distributed.html). 

I find that the data in Minio is periodically getting deleted. Is this due to some configuration in Mimir and can it be prevented? Or is it something related to Minio configuration?

**Comment by pracucci** [comment, marked as answer]
> I find that the data in Minio is periodically getting deleted. Is this due to some configuration in Mimir and can it be prevented? Or is it something related to Minio configuration?

The default Mimir configuration has retention disabled, which means data is never deleted. Minio shouldn't delete any data as well when running with the default configuration. However, the reason why you could see object deleted from Minio is because of the Mimir [compactor](https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/compactor/): small blocks are compacted into larger blocks, and then the small source blocks are deleted. Data is not actually delete, in the sense that you don't loose any metrics data.
